### PR TITLE
[Clingcon] Add builder

### PR DESCRIPTION
**Pull request by Wimmerer**
A little cart before the horse here given I haven't even finished the Clingo wrapper. Feel free to close since this won't be wrapped for at least 2 months.

### CODE REVIEW

File: C/Clingcon/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    -DPYCLINGCON_ENABLE=OFF \
+    -DCMAKE_CXX_FLAGS="-std=c++17" \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    ..
+cmake --build . --target install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline C/Clingcon/build_tarballs.jl:30]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: C/Clingcon/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    LibraryProduct("libclingcon",:libclingcon)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("Clingo_jll")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"8")
```
Comment (giordano) [inline C/Clingcon/build_tarballs.jl:44]: Can we require Julia v1.6?  It'll make it easier to add new platforms in the future
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"8", julia_compat="1.6")
```

Comment (Wimmerer) [reply C/Clingcon/build_tarballs.jl:44]: I don't see why not, I have no use for <1.6 myself. Should I be doing that by default now? For new builders?

Comment (giordano) [reply C/Clingcon/build_tarballs.jl:44]: That'd be good, yes, see https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/1047 and https://github.com/JuliaPackaging/Yggdrasil/issues/2763

**Pull request merged**